ROCm 6.0.0 and 6.0.2 releases use mix between LLVM 17 and 18
forked as https://github.com/RadeonOpenCompute/llvm-project
which makes some libraries compatible with only one version,
while other require another...

Backports https://github.com/ROCm/llvm-project/commit/6cbc4dc91dfeb1cf2295cb350866e0b3a07dfee4
--- a/src/comgr-metadata.cpp
+++ b/src/comgr-metadata.cpp
@@ -1087,7 +1087,12 @@ amd_comgr_status_t lookUpCodeObject(DataObject *DataP,
   }
 
   BinaryStreamReader Reader(StringRef(DataP->Data, DataP->Size),
-                            support::little);
+#if LLVM_VERSION_MAJOR > 17
+                            llvm::endianness::little
+#else
+                            support::little
+#endif
+                            );
 
   StringRef Magic;
   if (auto EC = Reader.readFixedString(Magic, OffloadBundleMagicLen)) {
